### PR TITLE
Extension loader load extension classes

### PR DESCRIPTION
**Pull request by dqqzj**
Check if type name starts with org.apache

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java [hunk 1]
```diff
@@ -747,14 +749,15 @@ private void cacheDefaultExtensionName() {
...
+            cachedDefaultName = value;
+           /* String[] names = NAME_SEPARATOR.split(value);
             if (names.length > 1) {
                 throw new IllegalStateException("More than 1 default extension name on extension " + type.getName()
                         + ": " + Arrays.toString(names));
             }
             if (names.length == 1) {
                 cachedDefaultName = names[0];
-            }
+            }*/
```
Comment (chickenlj) [inline dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java]: Why comment this snippet?

Comment (dqqzj) [reply dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java]: If there are multiple type names separated by commas, an error will be directly reported below, that is, multiple types cannot be displayed, so why do we need to use separators? Think of it as a type name

File: dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java [hunk 2]
```diff
@@ -727,10 +727,12 @@ private boolean isSetter(Method method) {
         cacheDefaultExtensionName();
 
         Map<String, Class<?>> extensionClasses = new HashMap<>();
-
+        String typeName = type.getName();
         for (LoadingStrategy strategy : strategies) {
-            loadDirectory(extensionClasses, strategy.directory(), type.getName(), strategy.preferExtensionClassLoader(), strategy.excludedPackages());
-            loadDirectory(extensionClasses, strategy.directory(), type.getName().replace("org.apache", "com.alibaba"), strategy.preferExtensionClassLoader(), strategy.excludedPackages());
```
Comment (dqqzj) [inline dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java]: If someone uses a lower version, org.apache will not be matched, and it is useless logic to perform the replacement.

Comment (AlbumenJ) [reply dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java:733]: In current, `org.apache` will be matched always, would this be roundabout?

**Comment by AlbumenJ** [comment]
Close for long time no response.

Please feel free to reopen if you have any question.

If you think these changes still useful in the latest master branch, please submit a new pull request.

**Pull request closed without merging**